crates/worker/src/actor/spawner.rs

```diff
@@ -176,8 +188,61 @@ where
         W::Input: Serialize + for<'de> Deserialize<'de>,
         W::Output: Serialize + for<'de> Deserialize<'de>,
     {
-        let worker = DedicatedWorker::new(loader_path).expect("failed to spawn worker");
+        self.spawn_with_options(SpawnOptions::new(loader_path).with_loader())
+            .expect("failed to spawn worker")
+    }
+
+    /// Spawns a Worker according to the given options. Returns `None` if failed.
+    pub fn spawn_with_options(&self, options: SpawnOptions<'_>) -> Option<WorkerBridge<W>>
+    where
+        W::Input: Serialize + for<'de> Deserialize<'de>,
+        W::Output: Serialize + for<'de> Deserialize<'de>,
+    {
+        let worker = create_worker(options)?;
 
-        self.spawn_inner(worker)
+        Some(self.spawn_inner(worker))
+    }
+}
+
+/// Options for spawning a worker.
+#[derive(Debug, Copy, Clone)]
+pub struct SpawnOptions<'a> {
```

Comment:
Using options to store this would hinder tree shaking as it would be impossible for the compiler to understand whether a loader / module is used at compile time.

I would prefer to make the functions separate as `spawn`, `spawn_with_loader`, and `spawn_as_module` so compiler can perform tree shaking on the shim script if a loader is used.

If you wish to keep this pattern, it should be moved to spawner types, so a builder pattern can be called like the following:

```rust
MyWorker::spawner()
    .with_loader()
    .as_module()
    .spawn("./my_worker.js");
```

Reply:
I had no idea about that, so I'm happy to go with whatever you think will make this work. I prefer the latter solution since otherwise if any more options are added the number of `spawn_*` options is exponential (i.e. we'd also need `spawn_with_loader_as_module`)

Reply:
How does my current implementation look? It's a bit annoying that `as_module` takes the `bool` argument while `with_loader` doesn't: they should probably be consistent.

Pros from having `bool` arg:
- The spawn function takes `&self` and so could be used to spawn multiple times: I could use it as e.g. if I try spawning and get an error (assuming `spawn` doesn't panic) then I toggle some settings and try again.
- I think `as_module` is a nicer name than `as_classic`.

Cons:
- More cumbersome

Reply:
I think both ways are fine.
However, I think it is important to make sure the API is consistent so that loader and module have same kind of schematics.